server/src/main/java/org/opensearch/action/search/TransportSearchAction.java

```diff
@@ -494,13 +494,12 @@ private void executeRequest(
         );
 
         final List<SearchRequestOperationsListener> searchListenersList = createSearchListenerList(originalSearchRequest, timeProvider);
-
-        final SearchRequestOperationsListener searchRequestOperationsListener;
-        if (!CollectionUtils.isEmpty(searchListenersList)) {
-            searchRequestOperationsListener = new SearchRequestOperationsListener.CompositeListener(searchListenersList, logger);
-        } else {
-            searchRequestOperationsListener = null;
-        }
+        final SearchRequestOperationsListener searchRequestOperationsListener = new SearchRequestOperationsListener.CompositeListener(
+            searchListenersList,
+            logger
+        );
+        SearchRequestContext searchRequestContext = new SearchRequestContext();
```

Comment:
Rather than adding *another* parameter that gets added to every subclass of `AbstractSearchAsyncAction`, I'm wondering if maybe it makes sense to make `searchRequestOperationsListener` a field of `SearchRequestContext`.

You still need to modify all the classes to switch from passing `searchRequestOperationsListener` to passing `searchRequestContext`, but at least you're replacing a parameter rather than adding another one.

Reply:
+1.

Also wondering whether we should also have SearchPhaseContext within RequestContext. So that we don't have to pass both in our listeners. 
As request comprises of phases, so kinda make sense.

Reply:
Sure that makes sense since `searchRequestOperationsListener` is a request-level composite listener